src/libcore/mem/mod.rs

```diff
@@ -69,8 +69,29 @@ pub use crate::intrinsics::transmute;
 /// ```
 ///
 /// The practical use cases for `forget` are rather specialized and mainly come
-/// up in unsafe or FFI code. However, [`ManuallyDrop`] is usually preferred
-/// for such cases, e.g.:
+/// up in unsafe or FFI code. For example:
+///
+/// ```
+/// use std::mem;
+///
+/// let mut v = vec![65, 122];
+/// // Build a `String` using the contents of `v`
+/// let s = unsafe { String::from_raw_parts(v.as_mut_ptr(), 2, v.capacity()) };
+/// // immediately leak `v` because its memory is now managed by `s`
+/// mem::forget(v);
```

Comment:
This is exactly the kind of thing where one should use `ManuallyDrop` to avoid touching `v` again after calling `as_mut_ptr()`.

So, maybe add a fat warning that this is an example for how to *not* use `mem::forget`? But then the intro makes little sense.

Reply:
Warning added.